File: array/add1_tolist.cpp
vector<int> solve(vector<int>& n) {
    int carry = 0 , pushval;
    vector<int> ans;
    for (int i = n.size() - 1; i >= 0; i--) {
        if (n[i] == 9) {
            if (carry != 0 || i == n.size() - 1) {
                pushval = 0;
                carry = 1;
            } else
                pushval = n[i];

        } else {
            if (i == n.size() - 1)
                pushval = n[i] + 1;
            else
                pushval = n[i] + carry;
            carry = 0;
        }
        ans.push_back(pushval);
    }
    if (carry != 0) ans.push_back(carry);
    reverse(ans.begin(), ans.end());
    return ans;
}

File: array/add1_tolist.java
class Solution {
    static ArrayList<Integer> increment(ArrayList<Integer> arr , int N) {
       
       ArrayList<Integer> res=new ArrayList<>();
       int carry=0;
       for(int i=N-1;i>=0;i--){
           int dig=arr.get(i);
           if(i==N-1){
               if(dig==9){
                   res.add(0);
                   carry=1;
               }
               else
                   res.add(dig+1);
           }
           else{
               dig=dig+carry;
               carry=dig/10;
               res.add(dig%10);
           }
       }
       if(carry!=0)
           res.add(carry);
       
       Collections.reverse(res);
       return res;
        
    }
};
